imagick/watermark.go

```diff
@@ -0,0 +1,59 @@
+package imagick
+
+import (
+	"github.com/journeymidnight/pipa/library"
+	"gopkg.in/gographics/imagick.v3/imagick"
+)
+
+const GRAVITY = imagick.GRAVITY_SOUTH_EAST
+
+type Watermark struct {
+	XMargin      int
+	YMargin      int
+	Gravity      imagick.GravityType
+	Transparency int
+	Picture      *imagick.MagickWand
+	Text         *Text
+}
+
+type Text struct {
+	text     string
+	textType string
+	color    string
+	front    string
+	fontSize int
+	shadow   int
+	rotate   int
+	fill     bool
+}
+
+func newWatermark() Watermark {
+	return Watermark{library.XMargin, library.YMargin, GRAVITY, library.Transparency, nil, new(Text)}
+}
+
+func (img *ImageWand) watermark(o Watermark) (err error) {
+	if o.Picture != nil {
+		err = img.MagickWand.CompositeImage(o.Picture, o.Picture.GetImageCompose(), true, o.XMargin, o.YMargin)
```

Comment:
You only need to pass the o parameter, and the abbreviation should not be o，m is better